src/api/modules/zkp/helpers/credentials.ts

```diff
@@ -13,23 +19,52 @@ export const getClaimOffer = async (userDid: string, claimTypeUrn: string) => {
   return data
 }
 
+/**
+ * This method simply returns type of schema,
+ * which is usually the element of {@link W3CCredential.type}
+ *
+ * and credentialSubject properties with their types and values (optional)
+ * @param schemaUrl
+ * @param credentialSubject
+ */
 export const loadAndParseCredentialSchema = async (
   schemaUrl: string,
   credentialSubject?: CredentialSubject,
-): Promise<{
-  key: string
-  type: string
-  value: string
-}> => {
-  const { data } = await fetcher.get<VCSchema>(schemaUrl)
-
-  const [key, { type }] = Object.entries(
-    omit(data?.properties.credentialSubject.properties, 'id'),
+): Promise<ParsedCredentialSchema> => {
+  const { data: schema } = await fetcher.get<VCSchema>(schemaUrl)
+
+  if (!schema?.$metadata.uris.jsonLdContext) throw new TypeError('Invalid schema')
+
+  const { data: properties } = await fetcher.get<JsonLdSchema>(schema.$metadata.uris.jsonLdContext)
+
+  const propertiesContext = properties?.['@context']?.[0]
+
+  if (!propertiesContext) throw new TypeError('Invalid schema')
```

Comment:
the same error message will be hard to debug

could we use more specific message about the context?